setuptools_scm/hg.py

```diff
@@ -8,13 +8,18 @@
 def _hg_tagdist_normalize_tagcommit(root, tag, dist, node):
     dirty = node.endswith('+')
     node = node.strip('+')
-    st = do('hg st --no-status --change %s' % str(node), root)
-
-    trace('normalize', locals())
-    if int(dist) == 1 and st == '.hgtags' and not dirty:
-        return meta(tag)
+    revset = ("(branch(.) and tag('{0}')::. and file('re:^(?!\.hgtags).*$')"
+              " - tag('{0}'))").format(str(tag))
```

Comment:
i wonder if instead of using str here, the string itself should use `{tag!r}` and the format call should use `tag=tag`